### PR TITLE
[spicedb] Initial schema with projects

### PR DESCRIPTION
**Pull request by easyCZ**
## Related Issue(s)

Fixes #

## How to test

https://play.authzed.com/s/hSwh3V53xCly/schema

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

File: install/installer/pkg/components/spicedb/data/schema.yaml [hunk 1]
```diff
@@ -6,42 +6,82 @@ schema: |-
...
+    // Membership is the union of all organization specific roles
+    permission membership = owner + member
+
+    // Operations on organizations
     permission organization_read = owner + member
     permission organization_write = owner
 
+    // // Operations on Organization's Members
     permission organization_members_read = owner + member
     permission organization_members_write = owner + member
```
Comment (svenefftinge) [inline install/installer/pkg/components/spicedb/data/schema.yaml:22]: I don't think we need the `organization_` prefix or should have this.

Comment (easyCZ) [reply install/installer/pkg/components/spicedb/data/schema.yaml:22]: Was intended to distinguish for example `group_members_read` etc

File: install/installer/pkg/components/spicedb/data/schema.yaml [hunk 2]
```diff
@@ -6,42 +6,82 @@ schema: |-
...
+    // Organization specific roles
     relation owner: user
     relation member: user
 
+    // Membership is the union of all organization specific roles
+    permission membership = owner + member
+
+    // Operations on organizations
     permission organization_read = owner + member
     permission organization_write = owner
```
[the remark this answers is not in the document]
Comment (svenefftinge) [reply install/installer/pkg/components/spicedb/data/schema.yaml:18]: we could also align on a naming pattern such as: `[create|edit|view|delete]_aspect`

[the remark this answers is not in the document]
Comment (easyCZ) [reply install/installer/pkg/components/spicedb/data/schema.yaml:18]: Sure, sounds good. I'll update the org/members ones and leave the rest out to be added once we approach adding perm checks for these

File: install/installer/pkg/components/spicedb/data/schema.yaml (hunk 1 above)
Comment (svenefftinge) [reply install/installer/pkg/components/spicedb/data/schema.yaml:22]: but whenever you access this permission you need to qualify it anyway, no?

**Review by svenefftinge — APPROVED** [review]
LGTM Can I somehow already run the validation on the command line?

**Comment by easyCZ** [comment]
> LGTM Can I somehow already run the validation on the command line?

https://github.com/gitpod-io/gitpod/actions/runs/5312674673/jobs/9617510287?pr=17973

**Comment by easyCZ** [comment]
Or 
```
 zed validate schema-and-assertions.zaml
```
By pointing to the schema.yaml file

File: install/installer/pkg/components/spicedb/data/schema.yaml [hunk 3]
```diff
@@ -6,57 +6,107 @@ schema: |-
...
 
-    permission organization_members_read = owner + member
-    permission organization_members_write = owner + member
+    // General operations on organization
+    permission read_info = owner + member
+    permission write_info = owner
+
+    // Operations on Organization's Members
+    permission read_members = owner + member
+    permission write_members = owner + member
```
Comment (geropl) [inline install/installer/pkg/components/spicedb/data/schema.yaml:22]: Sorry for being late here, but: Was this `+ member` intended? :thinking:

Comment (easyCZ) [reply install/installer/pkg/components/spicedb/data/schema.yaml:22]: We can change it, but currently any member of an org can invite others

Comment (geropl) [reply install/installer/pkg/components/spicedb/data/schema.yaml:22]: Ah, my bad: Everbody can create an invite link :+1:

**Pull request merged**